org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextValidator.java

```diff
@@ -1234,5 +1234,32 @@ public void checkOppositeReferenceUsed(Assignment assignment) {
 			}
 		}
 	}
-
+	
+	@Check
+	public void checkOverridingRule(AbstractRule rule) {
+		final String name = rule.getName();
+		
+		for (Grammar g : GrammarUtil.getGrammar(rule).getUsedGrammars()) {
+			final AbstractRule r = GrammarUtil.findRuleForName(g, rule.getName());
+			
+			if (r != null) {
+				if (!rule.isOverride()) {
+					warning("This rule overrides " + name + " in " + GrammarUtil.getGrammar(r).getName() + ".", rule,
+							XtextPackage.Literals.ABSTRACT_RULE__NAME, XtextConfigurableIssueCodes.EXPLICIT_OVERRIDE_MISSING);
+					break;
+					
+				} else {
+					// type compatibility checking is currently done in 'Xtext2EcoreTransformer' being called during linking
+					//  so it's omitted here
+				}
+				
+			} else {
+				if (rule.isOverride()) {
+					error("None of the used grammars contain a rule named " + name, rule,
```

Comment:
"used grammars" is not a user-level term. 
How about "The rule 'Foo' must override a rule from a super grammar"?